packages/launcher/lib/browsers.ts

```diff
@@ -181,7 +182,9 @@ export function launch (
 
   // allow setting default env vars such as MOZ_HEADLESS_WIDTH
   // but only if it's not already set by the environment
-  const env = Object.assign({}, defaultBrowserEnv, process.env)
+  const env = Object.assign({}, defaultBrowserEnv, browser.env, launchOptionsEnv, process.env)
```

Comment:
@ryanjduffy It looks like this change is doing more than accepting env values form the launchOptions. I wouldn't expect envs set on the found browser to be used in the node process that starts the browser. In the previous PR you had opened, you had linked to a few code snipped in reply & you're plugin. Where are these ENV values actually getting applied?

Reply:
Not sure I follow but I'll share some context here in hopes that clarifies!

Here's the current code in our plugin. It's of course not setting env variables yet because it can't! 

https://github.com/replayio/replay-cli/blob/c90c03f08d9ccfcb9bc9a4a6a7c07246ff1b07f5/packages/cypress/src/index.ts#L11-L14

The eventual implementation after this landed would be:

```js
  on("before:browser:launch", (browser, launchOptions) => {
    selectedBrowser = browser.family;
    reporter.onTestSuiteBegin(undefined, "CYPRESS_REPLAY_METADATA");

    launchOptions.env = {
      RECORD_ALL_CONTENT: 1,
      RECORD_REPLAY_METADATA_FILE: '/path/to/file.json'
    };

    return launchOptions;
  });
```

The goal here being that we need to configure the replay browser with `RECORD_ALL_CONTENT` to instruct the firefox fork to record everything (unlike the desktop experience in which the user clicks the Rec button) and where to find metadata for the recording which is how we link up the recording with the test result.

This is all happening from the plugin because that's the point at which we have the test context to add the metadata. IIRC, I also didn't find another way to set an environment variables for the launcher node process beyond requiring the user to set them before invoking cypress.

Hope that helps! Happy to have a quick video call to chat if that's useful too.

Reply:
Also, i'm in the cypress community discord as ryanjduffy if you want to chat there!

Reply:
@ryanjduffy This is great insights! Thank you!

Reply:
Looks like `browsers.env` was just cruft from my prior impl. @emilyrohrbough 's suggestions to drop those two diffs should do the trick!